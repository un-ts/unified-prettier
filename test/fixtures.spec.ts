import fs from 'node:fs/promises'
import path from 'node:path'

import { unified } from 'unified'

import { prettier } from 'unified-prettier'

it('should just work', async () => {
  const fixturesPath = path.resolve('test/fixtures')
  const files = await fs.readdir(fixturesPath)
  for (const file of files) {
    const filePath = path.resolve(fixturesPath, file)
    const vFile = await unified()
      .use(prettier)
      .process({
        path: filePath,
        value: await fs.readFile(filePath),
      })
    expect(vFile.value).toMatchSnapshot(file)
  }
})
