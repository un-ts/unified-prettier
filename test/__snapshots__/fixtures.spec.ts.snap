// Vitest Snapshot v1

exports[`it should just work > test.html 1`] = `
"<!DOCTYPE html>
<html>
  <head>
    <title>Hello World!</title>
  </head>
  <body></body>
</html>
"
`;

exports[`it should just work > test.md 1`] = `
"import {Demo} from 'demo';
export {Demo}

# Header

1. List item 1
2. List item 2
3. List item 3
"
`;

exports[`it should just work > test.mdx 1`] = `
"import { Demo } from 'demo'
export { Demo }

# Header

1. List item 1
2. List item 2
3. List item 3
"
`;
